### ISSUE TITLE
Should we use hmpo-model as the session model?

### ISSUE THREAD

**Issue by daniel-ac-martin** [UKHomeOffice/hof#95]
In the recent spirit of questioning HOF's entire architecture, is it worth thinking about whether we should be using hmpo-model for sessions? Isn't there a whole load of stuff in there for talking to RESTful APIs? Couldn't we have something simpler?

**Comment by JoeChapman** [comment]
There are occasions when we need to talk to RESTful API's.

**Comment by easternbloc** [comment]
> There are occasions when we need to talk to RESTful API's.

This is true although a more basic model might be valid for the hof core.

**Comment by daniel-ac-martin** [comment]
@JoeChapman: But do those occasions occur when using the session model?

**Comment by JoeChapman** [comment]
It might be, but is it a priority?

**Comment by easternbloc** [comment]
> It might be, but is it a priority?

No, but I think while we're questioning all the things it warrants a question. We need to have a list of all the stuff we think needs addressing and then we can just label them on a roadmap.

Sound good?

**Comment by JoeChapman** [comment]
That sounds fine. However, I actually think it's useful to have the RESTful API code. But maybe split them? I do however, think that because of the fresh look we are taking at HOF, that it might be in our interest to stop using other components in the HOF ecosystem, i.e. HMPO-wizard/HMPO-controller. Not least because they don't have the same obligation to ask for our feedback when making code changes to HMPO code that we do to HMPO code

**Comment by easternbloc** [comment]
> But maybe split them?

Yus! With regards to the other comments I think they're perfectly valid options and should be raised as further issues. We don't have to do everything that's raised as an issue but it's good to capture our thoughts and discuss them openly here.

**Comment by JoeChapman** [comment]
Didn't mean to sound argumentative. I should put smiley faces next to all my comments. Just so you know I love you @daniel-ac-martin :)

**Comment by easternbloc** [comment]
@JoeChapman ![](https://media.giphy.com/media/YXxmbqoUAta24/giphy.gif)
